src/main/java/com/uber/cadence/internal/shadowing/ReplayWorkflowActivityImpl.java

```diff
@@ -114,21 +111,22 @@ public ReplayWorkflowActivityResult replay(ReplayWorkflowActivityParams request)
       successCount += oneReplayResult.getSucceeded();
       failedCount += oneReplayResult.getFailed();
       skippedCount += oneReplayResult.getSkipped();
-      ReplayWorkflowActivityResult heartbeatResult =
-          new ReplayWorkflowActivityResult()
-              .setSucceeded(successCount)
-              .setFailed(failedCount)
-              .setSkipped(skippedCount);
+      ReplayWorkflowActivityResult heartbeatResult = new ReplayWorkflowActivityResult();
+      heartbeatResult.setSucceeded(successCount);
+      heartbeatResult.setFailed(failedCount);
+      heartbeatResult.setSkipped(skippedCount);
       Activity.heartbeat(new HeartbeatDetail(heartbeatResult, replayIndex));
     }
-    return new ReplayWorkflowActivityResult()
-        .setSucceeded(successCount)
-        .setFailed(failedCount)
-        .setSkipped(skippedCount);
+    ReplayWorkflowActivityResult result = new ReplayWorkflowActivityResult();
```

Comment:
nit: same here, looks like `heartbeatResult` can be reused.

Reply:
updated